### PR TITLE
Adding multimin.rs wrapping GSL's Multidimensional Minimization.

### PR DESCRIPTION
**Pull request by hugohp**
Adding multimin.rs,  wrapping GSL's Multidimensional Minimization

* Only functions without derivatives are supported now.
* GSL Multidimensional Minimization: https://www.gnu.org/software/gsl/doc/html/multimin.html
* What formatting tools do you use? I manually broke comment lines but I suspect you use some kind of tool? Which one?

### CODE REVIEW

File: src/types/multimin.rs [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    Minimizer<'a>,
+    *mut sys::gsl_multimin_fminimizer,
+    gsl_multimin_fminimizer_free
+    ;inner_call: sys::gsl_multimin_function_struct => sys::gsl_multimin_function_struct { f: None, n: 0_usize, params: std::ptr::null_mut() };
+    ;inner_closure: Option<Box<dyn Fn(&VectorF64) -> f64 + 'a>> => None;
+);
+
+impl<'a> Minimizer<'a> {
+    /// Creates a minimizer of type `t` for an n-dimensional function.
+	/// If there is insufficient memory to create the minimizer then
```
Comment (GuillaumeGomez) [inline src/types/multimin.rs:84]: The indent of the doc comment is off (same below). Please fix it.

File: src/types/multimin.rs [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        ffi_wrap!(gsl_multimin_fminimizer_nmsimplex2rand)
+    }
+}
+
+#[cfg(any(test, doctest))]
+mod test {
+    /// This doc block will be used to ensure that the closure can't be set everywhere!
+    ///
+    /// ```compile_fail
+	/// extern crate rgsl;
```
Comment (GuillaumeGomez) [inline src/types/multimin.rs:203]: I wonder though, did your IDE format like this? :laughing:

**Review by GuillaumeGomez — COMMENTED** [review]
Apart from the indent, looks good to me. Thanks for working on this! Fix the indent and then I'll merge.

File: src/types/multimin.rs (hunk 1 above)
Comment (hugohp) [reply src/types/multimin.rs]: Ok I replaced all \t with four-spaces. I am using vim.

File: src/types/multimin.rs (hunk 2 above)
Comment (hugohp) [reply src/types/multimin.rs]: yes I was mixturing tabs and spaces! I remove all tabs.

**Comment by hugohp** [comment]
I pushed a commit.

**Comment by GuillaumeGomez** [comment]
You forgot to run `cargo fmt` apparently.

**Comment by hugohp** [comment]
I have run it and pushed changes.

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**